### PR TITLE
[Bug](java-udf) fix java-udf memory leak

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
have new object twice, but only have one delete.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]
LGTM

File: be/src/vec/functions/function_java_udf.cpp [hunk 1]
```diff
@@ -85,39 +85,41 @@ Status JavaFunctionCall::open(FunctionContext* context, FunctionContext::Functio
...
-    {
-        std::string local_location;
-        auto function_cache = UserFunctionCache::instance();
-        RETURN_IF_ERROR(function_cache->get_jarpath(fn_.id, fn_.hdfs_location, fn_.checksum,
-                                                    &local_location));
-        TJavaUdfExecutorCtorParams ctor_params;
-        ctor_params.__set_fn(fn_);
-        ctor_params.__set_location(local_location);
-        jbyteArray ctor_params_bytes;
+    if (scope == FunctionContext::FunctionStateScope::THREAD_LOCAL) {
```
Comment (Gabriel39) [inline be/src/vec/functions/function_java_udf.cpp:89]: Use `else`

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**